File: src-tauri/src/handlers/cfg.rs
use crate::{config::Config, handlers::BackendError};

#[tauri::command]
pub async fn get_config() -> Result<Config, BackendError> {
	let cfg = Config::load();
	Ok(cfg)
}
// need to properly look at this later, most likely has cases where it fails and gives an exception
#[tauri::command]
pub async fn update_config(config: Config) -> Result<(), BackendError> {
	config.save();
	Ok(())
}

File: src/commands/cfg.ts
import { invoke } from "@tauri-apps/api/core";
import type { Config } from "../types/cfg";

export async function getConfig(): Promise<Config> {
  return invoke<Config>("get_config");
}

export async function updateConfig(config: Config): Promise<void> {
  return invoke("update_config", { config });
}


